nodes/config/ui_group.html

```diff
@@ -6,7 +6,11 @@
                 value: 'Group Name',
                 required: true
             },
-            page: { type: 'ui-page', required: true },
+            page: { type: 'ui-page', required: false },
+            path: {
```

Comment:
Thanks for the suggestion @idolieb26 - not sure I follow why we'd need a default path on a _group_ though. A single page contains many groups of widgets. All of them render at the same path (defined by the Page).

https://github.com/flowforge/flowforge-nr-dashboard/issues/137 is referring to adding this option at a `ui_base` level, and would allow you to select any existing `ui-page` elements.

Reply:
@joepavitt But there is already a `path` field in `ui-base`. So  I just need to select any existing `ui-page` elemensts, right?

Reply:
That's correct, so a default page being added to a `ui-base` would be the way to go. We have a four-tier hierarchy:

- Base: defines the `base` URL, e.g. `/dashboard/`
- Page: individual tabs/pages, all hosted at the same base, will extend the base URL, e.g. `/dashboard/page1`
- Group: All exist within a given page
- Widget: All exist within a given group

So if someone navigates to `/dashboard` or any subpath that isn't a valid page, we need to redirect them to _somewhere_. That somewhere is what we need to define as the `default`.

Two options exist for where this could live, at the `ui_base` level, in it's own configuration menu, or we have a global "Dashboard 2.0" side menu which controls global config options, where it could also live. This side menu is a WIP though in the branch I'm working on for https://github.com/flowforge/flowforge-nr-dashboard/issues/136

Reply:
@joepavitt I added `default` to `ui-base`, but there is no way to implement redirect logic in our dashboard, I guess.

Reply:
We have a VueJS app for the Dashboard. Node-RED is sending websocket events to the VueJS app. That app has a `router.js` which is empty by default, but then populated when we receive a `ui-config` socket event in `App.vue` which details the pages/routes.

Reply:
@joepavitt Yeah, that is what I guessed so. But the VueJS app is not in the `flowforge-nr-dashboard` project, right?

Reply:
Indeed it is - under `/ui`

Reply:
If you're running this locally, you'll need to run an `npm run build` from the cmd line at the project level, that builds an instance of the VueJS app, which Node-RED can then pick up. 

It's a pain to have to do a manual build step after each change, it's on our todo list to sort to also work with a dev server.